runtime/service/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -181,7 +181,7 @@ static int mapExceptionToResponseCode(RuntimeException rex) {
       case ForbiddenException e -> Status.FORBIDDEN.getStatusCode();
       case jakarta.ws.rs.ForbiddenException e -> Status.FORBIDDEN.getStatusCode();
       case NotAuthorizedException e -> Status.UNAUTHORIZED.getStatusCode();
-      case NamespaceNotEmptyException e -> Status.BAD_REQUEST.getStatusCode();
+      case NamespaceNotEmptyException e -> Status.CONFLICT.getStatusCode();
```

Comment:
How well does it work with older Iceberg java clients? What's their expectation in this failure mode?